lib/template-mixins.js

```diff
@@ -91,7 +100,7 @@ module.exports = function (t, fields, options) {
             error: this.errors && this.errors[key],
             maxlength: maxlength(key) || extension.maxlength,
             required: extension.required !== undefined ? extension.required : true,
-            pattern: extension.pattern
+            pattern: pattern(key) || extension.pattern
```

Comment:
On further thought I think this would need a little bit of refactoring to be able to `pattern: false` in config and have that be respected to disable the pattern.